src/main/resources/public/jsx/geo-dash-widget-editor.js

```diff
@@ -774,7 +762,7 @@ class BasicLayout extends React.PureComponent{
         }
         else {
             this.setState({endDateDual: date});
```

Comment:
checkDates should return true or false instead of changing the state. 
```this.setState({endDateDual: date,
   FormReady: this.checkDates()
});

Reply:
currently isFormReady has 3 possibilities null (leave state alone), true (set state to ready), false (set state to not ready)